contracts/rewards/Reward.sol

```diff
@@ -46,10 +46,20 @@ abstract contract Reward is IReward, ERC2771Context, ReentrancyGuard {
   /// @inheritdoc IReward
   uint256 public supplyNumCheckpoints;
 
+  /// @inheritdoc IReward
+  address public gauge;
+
   constructor(address _forwarder, address _voter) ERC2771Context(_forwarder) {
     voter = _voter;
   }
 
+  function setGauge(address _gauge) external {
+    require(gauge == address (0), "gauge has been set");
+    if (msg.sender != authorized) revert NotAuthorized();
```

Comment:
authorized is not initialized

Reply:
initialized in CustomReward when construct.